### DISCUSSION TITLE
LTSP clients not finding server

### DISCUSSION THREAD

**Discussion by decenttp** [ltsp/ltsp#512] [Q&A] [answered]
Hi,

I recently installed new LTSP on Ubuntu 20.04 LTS and absolutely loved it. Got it up and running in no time. However, after i upgraded yesterday something seems to have broken and the clients can no longer find the server or the boot image. I have a basic setup with an external DHCP server (Also an internet gateway). Both DHCP and LTSP servers are in the same network range 192.168.1.0 and LTSP server even though it has 2 network cards, has only one configured as of now. I am trying to first fix what could have caused the issue but lost right now and would highly appreciate some pointers. I have attached the current ltsp.conf and ltsp.ipxe. 

Thanks a lot. 

[ltsp.conf.txt](https://github.com/ltsp/ltsp/files/6980214/ltsp.conf.txt)
[ltsp.ipxe.txt](https://github.com/ltsp/ltsp/files/6980231/ltsp.ipxe.txt)

**Comment by alkisg** [comment]
Run journalctl -f on the ltsp server and try to boot a client. Upload the server terminal output and a photo of the client screen.

Also upload your /etc/dnsmasq.d/ltsp-dnsmasq.conf.

**Comment by decenttp** [comment]
Thanks for your response. Please find the details you asked for. 
[ltsp-dnsmasq.conf.txt](https://github.com/ltsp/ltsp/files/6980488/ltsp-dnsmasq.conf.txt)

![processed](https://user-images.githubusercontent.com/88873876/129316867-05e01bd4-909c-4fec-9394-3cd751ebc3cb.jpeg)
![processed2](https://user-images.githubusercontent.com/88873876/129316870-185f9b97-957b-40ce-b08d-b8eac0d2d447.jpeg)
[journalctl.txt](https://github.com/ltsp/ltsp/files/6980471/journalctl.txt)

**Comment by decenttp** [comment]
Not sure if this is of any help but still sharing output for ltsp info: 

root@vpn:/var/log# ltsp info
LTSP version: 21.01-1~ubuntu20.04.1

SERVER XSESSIONS: lxqt.desktop  ubuntu.desktop  xfce.desktop  xubuntu.desktop

SERVER OS: PRETTY_NAME="Ubuntu 20.04.2 LTS"

CHROOTS:


VMs:


IMAGES:
x86_64

**Comment by decenttp** [comment]
Just to confirm if DHCP server is accessible and assigning IP's i tested on a virtualbox client and that part seems to be working. 

![image](https://user-images.githubusercontent.com/88873876/129318104-2b302fbf-e92d-4c10-8b7c-ca6f48ba2edc.png)

**Comment by alkisg** [comment]
It looks like the client DHCP requests don't reach the LTSP server. Check its networking.

**Reply by decenttp** [reply]
Yes, when you asked for the information I realized the same but it's not just one client but all of them having the same issue. The virtualbox client was running on the same pc from where I had logged into the LTSP server via SSH.

**Reply by decenttp** [reply]
I misunderstood your comment. Checking networking on the LTSP. Like i said its certainly available on the network and i am connected remotely to it via ssh.

**Comment by alkisg** [comment]
Sorry I'm typing from phone I can't type much. Is dnsmasq running? journalctl -fu dnsmasq. If yes, check if the DHCP requests reach the LTSP server, not networking in general. E.g. if the server is directly connected to the router, instead of a switch, client DHCP requests may not reach the server.

**Reply by decenttp** [reply]
dnsmasq seems to be working :

Aug 12 19:05:47 vpn.lanserver.com dnsmasq[22774]: dnsmasq: syntax check OK.
Aug 12 19:05:47 vpn.lanserver.com dnsmasq[22783]: started, version 2.80 DNS disabled
Aug 12 19:05:47 vpn.lanserver.com dnsmasq[22783]: compile time options: IPv6 GNU-getopt DBus i18n IDN DHCP DHCPv6 no-Lua TFTP conntrack ipset auth nettlehash DNSSEC loop-detect inotify dumpfile
Aug 12 19:05:47 vpn.lanserver.com dnsmasq-dhcp[22783]: DHCP, proxy on subnet 192.168.1.0
Aug 12 19:05:47 vpn.lanserver.com dnsmasq-dhcp[22783]: DHCP, IP range 192.168.67.20 -- 192.168.67.250, lease time 12h
Aug 12 19:05:47 vpn.lanserver.com dnsmasq-tftp[22783]: TFTP root is /srv/tftp
Aug 12 19:05:47 vpn.lanserver.com systemd[1]: Started dnsmasq - A lightweight DHCP and caching DNS server.

LTSP server is connected to a switch and so is the router/DHCP server so it's not directly connected to the router. 

Based on what i read on the ltsp.org installation page : this usually means that you have a single NIC on the LTSP server and an external DHCP server, for example a router, pfsense, or a Windows server. In this case, run the following command:
# ltsp dnsmasq

I was following this.

**Comment by alkisg** [comment]
Your LTSP configuration seems OK, yet the dnsmasq logs do not contain any client DHCP requests. Are there any firewalls active?

**Reply by decenttp** [reply]
This router is a new fortigate device we just got and I am looking into the same now. Thanks a lot for your help thus far. I thought since both are connected to the switch, requests would be going directly instead of going through it.

**Reply by decenttp** [reply]
I just wanted to update incase anyone is having similar issues. Problem was related to the fortigate firewall only. It requires additional configuration through CLI to let clients find the server on the network.

I am facing a different problem however, with a separate configuration that tried in order to build a fat client setup. The ltsp server is Zorin 15.3 pc with a single network card. I  built an image using following parameters :

**sudo ltsp-build-client --dist bionic --chroot amd64mint --purge-chroot --fat-client --fat-client-desktop xfce4  --arch amd64** 

Had tried simple sudo ltsp-build-client --arch amd64 too but in every case i get following error (journalctl -f output) on the server

Sep 07 12:21:45 lan-ltsp dnsmasq-dhcp[22749]: DHCPDISCOVER(eno1) 08:00:27:e4:84:82
Sep 07 12:21:45 lan-ltsp dnsmasq-dhcp[22749]: DHCPOFFER(eno1) 192.168.1.190 08:00:27:e4:84:82
Sep 07 12:21:45 lan-ltsp dnsmasq-dhcp[22749]: DHCPREQUEST(eno1) 192.168.1.190 08:00:27:e4:84:82
Sep 07 12:21:45 lan-ltsp dnsmasq-dhcp[22749]: DHCPACK(eno1) 192.168.1.190 08:00:27:e4:84:82
Sep 07 12:21:45 lan-ltsp dnsmasq-dhcp[22749]: PXE(eno1) 192.168.1.190 08:00:27:e4:84:82 /ltsp/amd64mint/pxelinux.0
Sep 07 12:21:45 lan-ltsp dnsmasq-tftp[22749]: sent /var/lib/tftpboot/ltsp/amd64mint/pxelinux.0 to 192.168.1.190
Sep 07 12:21:46 lan-ltsp dnsmasq-tftp[22749]: sent /var/lib/tftpboot/ltsp/amd64mint/ldlinux.c32 to 192.168.1.190
Sep 07 12:21:46 lan-ltsp dnsmasq-tftp[22749]: file /var/lib/tftpboot/ltsp/amd64mint/pxelinux.cfg/bcef9eff-3268-4344-a732-bd3996678360 not found
Sep 07 12:21:46 lan-ltsp dnsmasq-tftp[22749]: file /var/lib/tftpboot/ltsp/amd64mint/pxelinux.cfg/01-08-00-27-e4-84-82 not found
Sep 07 12:21:46 lan-ltsp dnsmasq-tftp[22749]: file /var/lib/tftpboot/ltsp/amd64mint/pxelinux.cfg/C0A801BE not found
Sep 07 12:21:46 lan-ltsp dnsmasq-tftp[22749]: file /var/lib/tftpboot/ltsp/amd64mint/pxelinux.cfg/C0A801B not found
Sep 07 12:21:46 lan-ltsp dnsmasq-tftp[22749]: file /var/lib/tftpboot/ltsp/amd64mint/pxelinux.cfg/C0A801 not found
Sep 07 12:21:46 lan-ltsp dnsmasq-tftp[22749]: file /var/lib/tftpboot/ltsp/amd64mint/pxelinux.cfg/C0A80 not found
Sep 07 12:21:46 lan-ltsp dnsmasq-tftp[22749]: file /var/lib/tftpboot/ltsp/amd64mint/pxelinux.cfg/C0A8 not found
Sep 07 12:21:46 lan-ltsp dnsmasq-tftp[22749]: file /var/lib/tftpboot/ltsp/amd64mint/pxelinux.cfg/C0A not found
Sep 07 12:21:46 lan-ltsp dnsmasq-tftp[22749]: file /var/lib/tftpboot/ltsp/amd64mint/pxelinux.cfg/C0 not found
Sep 07 12:21:46 lan-ltsp dnsmasq-tftp[22749]: file /var/lib/tftpboot/ltsp/amd64mint/pxelinux.cfg/C not found
Sep 07 12:21:46 lan-ltsp dnsmasq-tftp[22749]: sent /var/lib/tftpboot/ltsp/amd64mint/pxelinux.cfg/default to 192.168.1.190
 
I would highly appreciate any pointers.

**Comment by alkisg** [comment, marked as answer]
ltsp-build-client is a tool of the older LTSP 5 and it's not supported anymore.

Also, for different topics, please open new discussions...
